pkg/watch/health.go

```diff
@@ -85,44 +101,48 @@ func MonitorPodHealth(config *preparer.PreparerConfig, logger *logging.Logger, s
 // service it is monitoring. Every HEALTHCHECK_INTERVAL it
 // performs a health check and writes that information to
 // consul
-func (p *PodWatch) MonitorHealth(node string, store kp.Store, shutdownCh chan bool) {
+func (p *PodWatch) MonitorHealth(store kp.Store, statusChecker StatusCheck, shutdownCh chan bool) {
 	for {
 		select {
 		case <-time.After(HEALTHCHECK_INTERVAL):
-			p.checkHealth(node, p.manifest.StatusPort, store)
+			p.checkHealth(store, statusChecker)
 		case <-shutdownCh:
 			return
 		}
 	}
 }
 
-func (p *PodWatch) checkHealth(node string, port int, store kp.Store) {
-	resp, err := kp.HttpStatusCheck(node, port)
+func (p *PodWatch) checkHealth(store kp.Store, sc StatusCheck) {
+	resp, err := sc.Check()
 	health, err := resultFromCheck(resp, err)
 	if err != nil {
 		return
 	}
-	health.ID = p.manifest.Id
-	health.Node = node
-	health.Service = p.manifest.Id
+	health.ID = sc.ID
```

Comment:
ret := foo.Check()
    ret.X = foo.X
    ret.Y = foo.Y
    return ret

This is not a good pattern to see. If the check is going to return a health result, the check itself should fill in as much of the result as it can, not rely on the caller to copy data between structs. In general, the more extra steps are required to use an API correctly, the greater the chance of misusing it. I'd like to see the check be as self contained as possible.

    return foo.Check()